src/axom/sidre/tests/sidre_view.cpp

```diff
@@ -1852,6 +1852,119 @@ TEST(sidre_view, deep_copy_shape)
 
 //------------------------------------------------------------------------------
 
+TEST(sidre_view, reshape_array)
+{
+  using namespace axom;
+  DataStore ds;
+  Group* root = ds.getRoot();
+
+  constexpr int DMAX = 4;
+  IndexType shapeOutput[DMAX];
+  int nDim = -1;
+
+  // A view with array in a buffer.
+  auto* viewA =
+    root->createView("viewA", sidre::detail::SidreTT<int32_t>::id, 12);
+
+  {
+    nDim = viewA->getShape(DMAX, shapeOutput);
+    EXPECT_EQ(viewA->getNumElements(), 12);
+    EXPECT_EQ(nDim, 1);
+    EXPECT_EQ(shapeOutput[0], 12);
+  }
+
+  {
+    // Attempt to reshape empty View should be no-op.
+    EXPECT_TRUE(viewA->isEmpty());
+    IndexType badShape[] = {1, 2, 3};
+    viewA->reshapeArray(3, badShape);
+    nDim = viewA->getShape(DMAX, shapeOutput);
+    EXPECT_EQ(viewA->getNumElements(), 12);
+    EXPECT_EQ(nDim, 1);
+    EXPECT_EQ(shapeOutput[0], 12);
+  }
+
+  viewA->allocate();
+  EXPECT_FALSE(viewA->isEmpty());
+  EXPECT_FALSE(viewA->isScalar());
+  EXPECT_TRUE(viewA->hasBuffer());
+
+  {
+    // Attempt to change size should be no-op.
+    IndexType badShape[] = {1, 2, 3};
+    viewA->reshapeArray(3, badShape);
+    nDim = viewA->getShape(DMAX, shapeOutput);
+    EXPECT_EQ(viewA->getNumElements(), 12);
+    EXPECT_EQ(nDim, 1);
+    EXPECT_EQ(shapeOutput[0], 12);
+  }
+
+  {
+    IndexType shape2d[] = {3, 4};
+    viewA->reshapeArray(2, shape2d);
+    nDim = viewA->getShape(DMAX, shapeOutput);
+    EXPECT_EQ(viewA->getNumElements(), 12);
+    EXPECT_EQ(nDim, 2);
+    EXPECT_EQ(shapeOutput[0], 3);
+    EXPECT_EQ(shapeOutput[1], 4);
+  }
+
+  viewA->deallocate();
+
+  {
+    IndexType shape2d[] = {2, 6};
+    viewA->reshapeArray(2, shape2d);
+    nDim = viewA->getShape(DMAX, shapeOutput);
+    EXPECT_EQ(viewA->getNumElements(), 12);
+    EXPECT_EQ(nDim, 2);
+    EXPECT_EQ(shapeOutput[0], 2);
+    EXPECT_EQ(shapeOutput[1], 6);
+  }
+
+  viewA->allocate();
```

Comment:
Can you please add a brief comment about how the `deallocate`/`allocate` calls affect the test?
E.g. if this block is testing the operation of `reshape` w/ an unallocated array.

Reply:
Good idea.  Done.